paddle/fluid/framework/parallel_executor.cc

```diff
@@ -396,20 +397,29 @@ ir::Graph *ParallelExecutorPrivate::ApplyMemoryOptimizePass(ir::Graph *graph) {
             BOOST_GET_CONST(platform::CUDAPlace, place), max_memory_size));
       }
       VLOG(10) << "Created " << i << "-th GarbageCollector at " << place;
-    } else {
+#else
+      PADDLE_THROW(platform::errors::PermissionDenied(
+          "Cannot use CUDAPlace in XPU/GPU version, "
```

Comment:
XPU/CPU

Reply:
`Paddle can't use CUDA device since it's not compiled with CUDA.`

Reply:
done!